src/scripts/main.js

```diff
@@ -1,3 +1,290 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('table');
+
+createForm();
+
+const form = document.querySelector('.new-employee-form');
+const tbody = document.querySelector('tbody');
+
+form.addEventListener('submit', function (e) {
+  // щоб не було перезавантаження при сабміті
+  e.preventDefault();
+
+  const res = checkForm();
+
+  if (res.type === 'error') {
+    pushNotification(10, 10, 'Error', res.message, 'error');
+  } else {
+    pushNotification(10, 10, 'Success', res.message, 'success');
+    createNewEmploye(form.elements);
+  }
+});
+
+table.addEventListener('dblclick', function (e) {
+  const target = e.target.closest('td');
+
+  if (!target) {
+    return; // якщо клікнуто не по комірці нічо не робити
+  }
+
+  const cellContent = target.textContent;
+
+  const input = createInput();
+
+  function createInput() {
+    const inputCtrl = document.createElement('input');
+
+    inputCtrl.className = 'cell-input';
+    inputCtrl.type = 'text';
+    target.innerHTML = '';
+    target.append(inputCtrl);
+    inputCtrl.focus();
+
+    return inputCtrl;
+  }
+
+  function deleteInput() {
+    if (input.value === '') {
+      target.textContent = cellContent;
+    }
+
+    if (!checkInput(cellContent, input)) {
+      pushNotification(10, 10, 'Error', 'Incorect type value', 'error');
+      input.focus();
+
+      return;
+    }
+
+    if (input.value !== '') {
+      target.textContent = input.value;
+    }
+    input.remove();
+  }
+
+  input.onblur = function (ev) {
+    deleteInput();
+  };
+
+  document.onkeyup = function (ev) {
+    if (ev.key === 'Enter') {
+      deleteInput();
+    }
+  };
+});
+
+table.addEventListener('click', function (e) {
+  const target = e.target.closest('th, td');
+  const targetRow = e.target.closest('tr');
+
+  if (!target && !targetRow) {
+    return undefined;
+  }
+
+  const rows = Array.from(table.querySelectorAll('tr'));
+
+  if (targetRow && targetRow.matches('tr')) {
+    singleActivateRow(rows, targetRow.rowIndex);
+  }
+
+  if (target && target.matches('th') && target.dataset.type) {
+    const columns = Array.from(tbody.rows).map(
+      (el) => el.cells[target.cellIndex],
+    );
+    let order = target.dataset.order === 'asc' ? 'desc' : 'asc';
+
+    order = target.dataset.old ? order : 'asc';
+
+    target.setAttribute('data-order', order);
+
+    // якщо перед цим був порядок 'asc'
+    // то міняю результат порівняння на від'ємний,
+    // щоб змінити порядок на 'desc' і навпаки
```

Comment:
it is not good practice to live comments.
code should be clean and "self commented".

if you need comments - general convention to leave them in english